src/AppInstallerCLICore/Commands/UpgradeCommand.cpp

```diff
@@ -19,8 +19,19 @@ namespace AppInstaller::CLI
     {
         bool ShouldListUpgrade(Context& context)
         {
-            return context.Args.Empty() ||
-                (context.Args.GetArgsCount() == 1 && context.Args.Contains(Execution::Args::Type::Source));
+            if (!context.Args.Empty())

```

Comment:
I don't think you need to check for empty here. If there isn't any, it should immediately go past the `for` and return true.

Reply:
Fixed in latest commit!